src/init.cpp

```diff
@@ -1046,8 +1048,15 @@ bool AppInitParameterInteraction()
     }
 
     nConnectTimeout = gArgs.GetArg("-timeout", DEFAULT_CONNECT_TIMEOUT);
-    if (nConnectTimeout <= 0)
+    if (nConnectTimeout <= 0) {
         nConnectTimeout = DEFAULT_CONNECT_TIMEOUT;
+    }
+
+    int64_t peer_connect_timeout_arg = gArgs.GetArg("-peertimeout", DEFAULT_PEER_CONNECT_TIMEOUT);
+    if (peer_connect_timeout_arg <= 0) {
+        return InitError(_("peertimeout cannot be configured with a negative value."));
+    }
+    peer_connect_timeout = (uint64_t) peer_connect_timeout_arg;
```

Comment:
(1/3) It would seem the insistence on making `peer_connect_timeout` a `uint64_t` is making the code a tiny more verbose (compared the block here with the one above for `-timeout`).